Source/control.cpp

```diff
@@ -1574,7 +1651,7 @@ void CheckChrBtns()
 			continue;
 		}
 		auto buttonId = static_cast<size_t>(attribute);
-		if (ChrBtnsRect[buttonId].Contains({ MousePosition.x - LEFT_PANEL_X, MousePosition.y - LEFT_PANEL_Y })) {
+		if (ChrBtnsRect[buttonId].Contains({ MousePosition.x - LeftPanel.position.x, MousePosition.y - LeftPanel.position.y })) {

```

Comment:
It seams a bit non nonsensical to move the mouse position rather then the button.

```suggestion
		Rectangle button = ChrBtnsRect[buttonId];
		button.position = GetPanelPosition(UiPanels::Character, button.position);
		if (button.Contains(MousePosition)) {
```
Same for line 1670

Also it looks like you also fully missed this adjustment in `AttrIncBtnSnap()` so it's probably not possible to level up using a gamepad atm.

Reply:
The gamepad issue has been fixed.